### PR TITLE
implement add_time_string_null

### PR DESCRIPTION
**Pull request by summer-slow-grace**
## What have you changed? (mandatory)
This PR implements the scalar function AddTimeStringNull
## What are the type of the changes? (mandatory)
New feature (non-breaking change which adds functionality)
## How has this PR been tested? (mandatory)
unit test
## Does this PR affect documentation (docs) or release note? (mandatory)
no
## Does this PR affect tidb-ansible update? (mandatory)
no
## Refer to a related PR or issue link (optional)
#3275

### CODE REVIEW

File: src/coprocessor/dag/expr/builtin_time.rs [hunk 1]
```diff
@@ -1701,4 +1710,10 @@ mod tests {
         test_ok_case_zero_arg(&mut ctx, ScalarFuncSig::AddTimeDurationNull, Datum::Null);
         test_ok_case_zero_arg(&mut ctx, ScalarFuncSig::SubTimeDurationNull, Datum::Null);
     }
+
+    #[test]
+    fn test_add_sub_time_string_null() {
```
Comment (rleungx) [inline src/coprocessor/dag/expr/builtin_time.rs]: Would you like to implement `SubTimeStringNull`?

Comment (summer-slow-grace) [reply src/coprocessor/dag/expr/builtin_time.rs]: A PR can only implement one function，So I only implement AddTimeStringNull,All test used test_add_sub as it's prefix,so i used the test_add_sub even though there is no sub function.

**Comment by summer-slow-grace** [comment]
@AndreMouche PTAL

**Comment by breeswish** [comment]
Thanks a lot!

**Comment by rleungx** [comment]
Merged, thx!

**Pull request merged**